src/panels/config/integrations/integration-panels/zwave_js/zwave_js-node-config.ts

```diff
@@ -0,0 +1,421 @@
+import { mdiRefresh } from "@mdi/js";
+import {
+  css,
+  CSSResultArray,
+  customElement,
+  html,
+  internalProperty,
+  LitElement,
+  property,
+  PropertyValues,
+  TemplateResult,
+} from "lit-element";
+import { debounce } from "../../../../../common/util/debounce";
+import "../../../../../components/ha-card";
+import "../../../../../components/ha-svg-icon";
+import "../../../../../components/ha-icon-next";
+import "../../../../../components/ha-switch";
+import {
+  fetchNodeConfigParameters,
+  setNodeConfigParameter,
+  ZWaveJSNodeConfigParams,
+} from "../../../../../data/zwave_js";
+import "../../../../../layouts/hass-tabs-subpage";
+import { haStyle } from "../../../../../resources/styles";
+import type { HomeAssistant, Route } from "../../../../../types";
+import "../../../ha-config-section";
+import { configTabs } from "./zwave_js-config-router";
+import {
+  DeviceRegistryEntry,
+  computeDeviceName,
+  subscribeDeviceRegistry,
+} from "../../../../../data/device_registry";
+import { SubscribeMixin } from "../../../../../mixins/subscribe-mixin";
+import { UnsubscribeFunc } from "home-assistant-js-websocket";
+import memoizeOne from "memoize-one";
+
+const getDevice = memoizeOne(
+  (entries: DeviceRegistryEntry[], deviceId: string) =>
+    entries.find((device) => device.id === deviceId)
+);
+
+const getIdentifier = memoizeOne((device: DeviceRegistryEntry) =>
+  device.identifiers.find((ident) => ident[0] === "zwave_js")
+);
+
+@customElement("zwave_js-node-config")
+class ZWaveJSNodeConfig extends SubscribeMixin(LitElement) {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ attribute: false }) public route!: Route;
+
+  @property({ type: Boolean }) public narrow!: boolean;
+
+  @property({ type: Boolean }) public isWide!: boolean;
+
+  @property() public configEntryId?: string;
+
+  @property() public deviceId!: string;
+
+  @property({ type: Array })
+  private _deviceRegistryEntries?: DeviceRegistryEntry[];
+
+  @internalProperty() private _config?: ZWaveJSNodeConfigParams[];
+
+  @internalProperty() private _error?: string;
+
+  public connectedCallback(): void {
+    super.connectedCallback();
+    this.deviceId = this.route.path.substr(1);
+  }
+
+  public hassSubscribe(): UnsubscribeFunc[] {
+    return [
+      subscribeDeviceRegistry(this.hass.connection, (entries) => {
+        this._deviceRegistryEntries = entries;
+      }),
+    ];
+  }
+
+  protected updated(changedProps: PropertyValues): void {
+    if (changedProps.has("_deviceRegistryEntries")) {
+      this._fetchData();
+    }
+  }
+
+  protected render(): TemplateResult {
+    if (this._error) {
+      return html`<hass-error-screen
+        .hass=${this.hass}
+        .error=${this.hass.localize(
+          `ui.panel.config.zwave_js.node_config.error_${this._error}`
+        )}
+      ></hass-error-screen>`;
+    }
+
+    if (!this._config) {
+      return html`<hass-loading-screen></hass-loading-screen>`;
+    }
+
```

Comment:
```suggestion
const device = this._device;
```

Or just use a memoize directly without the getter:

```ts
const device = getDevice(this._deviceRegistryEntries, this.deviceId);
```